Formula/zrepl.rb

```diff
@@ -0,0 +1,84 @@
+class Zrepl < Formula
+  desc "One-stop ZFS backup & replication solution"
+  homepage "https://zrepl.github.io"
+  url "https://github.com/zrepl/zrepl/archive/v0.1.1.tar.gz"
+  sha256 "0c16554e4527d14a390d78cf95bce759da425019a83ec63acfed5b4c50d68c9c"
+  head "https://github.com/zrepl/zrepl.git"
+
+  depends_on "dep" => :build
+  depends_on "go" => :build
+  def install
+    contents = Dir["{*,.git,.gitignore}"]
+    gopath = buildpath/"gopath"
+    (gopath/"src/github.com/zrepl/zrepl").install contents
+
+    ENV["GOPATH"] = gopath
+    ENV.prepend_create_path "PATH", gopath/"bin"    
+    cd gopath/"src/github.com/zrepl/zrepl" do
+      system "dep", "ensure", "-v", "-vendor-only"
+      system "go", "build", "-o", "'$GOPATH/bin/stringer'", "./vendor/golang.org/x/tools/cmd/stringer"
+      system "go", "build", "-o", "'$GOPATH/bin/protoc-gen-go'", "./vendor/github.com/golang/protobuf/protoc-gen-go"
+      system "go", "build", "-o", "'$GOPATH/bin/enumer'", "./vendor/github.com/alvaroloes/enumer"
+      system "go", "build", "-o", "'$GOPATH/bin/goimports'", "./vendor/golang.org/x/tools/cmd/goimports"
+      system "go", "build", "-o", "'$GOPATH/bin/golangci-lint'", "./vendor/github.com/golangci/golangci-lint/cmd/golangci-lint"
+      system "make", "ZREPL_VERSION=#{version}"
+      bin.install "artifacts/zrepl"
+    end      
+  end
+
+  def post_install
```

Comment:
Has an issue been filed for these missing paths? This looks like something upstream should generate.

Reply:
From `https://github.com/zrepl/zrepl/blob/master/docs/installation.rst`:

```
Note

It is your job to install the apropriate binary in the zrepl users's $PATH, e.g. /usr/local/bin/zrepl. Otherwise, the examples in the :ref:`tutorial` may need to be adjusted.
```

And furthermore documented here: https://zrepl.github.io/configuration/preface.html#runtime-directories

_zrepl requires runtime directories for various UNIX sockets — they are documented in the config file. Your __package maintainer / init script__ should take care of creating them. Alternatively, for default settings, the following should to the trick._

that's why I create them (and they may slightly differ between different platforms as well).

Reply:
That's a very shitty approach. But if you can't do anything about it I guess it's fine.

Reply:
why is that a shitty approach? The author is uaully very receptive to suggestions for change so if I can point him to some arguments why this approach is not ideal, please let me know

Reply:
Generally you want all installs to be roughly the same. The location and dependencies might differ but for most tools there will just be a bin directory, a /var/log directory and some other defaults that you can customize. Having worked some more with go recently, this might not be a very Go thing to do, but it's pretty nice for package managers.

Reply:
I've passed it on. Can we proceed as is with the formula for now?

Reply:
zrepl author here: what exactly is "shitty" to you?
The daemon / client design makes sense, and the client (cli) somehow needs to communicate to the daemon. UNIX socket is the obvious choice.
If any path (like `/var/run/zrepl`) doesn't fit, just patch the default path in homebrew (`grep -r '/var/run' path/to/zrepl/checkout`) or open a PR to make it configurable at build time ( see Makefile zrepl version tag for an example).